pkgs/test/pkg-config-packages.nix

```diff
@@ -0,0 +1,81 @@
+{ lib, pkg-config, pkg-configPackages, runCommand }:
+let
+  inherit (lib.strings) escapeNixIdentifier;
+
+  allTests = lib.mapAttrs (k: v: if v == null then null else makePkgConfigTestMaybe k v) pkg-configPackages;
+
+  # nix-build rejects attribute names with periods
+  # This will build those regardless.
+  tests-combined = runCommand "pkg-config-checks" {
+    allTests = lib.attrValues allTests;
+  } ''
+    touch $out
+  '';
+
+  makePkgConfigTestMaybe = moduleName: pkg:
+    if ! lib.isDerivation pkg
+    then
+      throw "pkg-config module `${escapeNixIdentifier moduleName}` is not defined to be a derivation. Please check the attribute value for `${escapeNixIdentifier moduleName}` in `pkgs/top-level/pkg-config-packages.nix` in Nixpkgs."
+
+    else if ! pkg?meta.unsupported
+    then
+      throw "pkg-config module `${escapeNixIdentifier moduleName}` does not have a `meta.unsupported` attribute. This can't be right. Please check the attribute value for `${escapeNixIdentifier moduleName}` in `pkgs/top-level/pkg-config-packages.nix` in Nixpkgs."
+
+    else if pkg.meta.unsupported
+    then
+      # We return `null` instead of doing a `filterAttrs`, because with
+      # `filterAttrs` the evaluator would not be able to return the attribute
+      # set without first evaluating all of the attribute _values_. This would
+      # be rather expensive, and severly slow down the use case of getting a
+      # single test, which we want to do in `passthru.tests`, or interactively.
+      null
+
+    else if ! pkg?meta.broken
+    then
+      throw "pkg-config module `${escapeNixIdentifier moduleName}` does not have a `meta.broken` attribute. This can't be right. Please check the attribute value for `${escapeNixIdentifier moduleName}` in `pkgs/top-level/pkg-config.packages.nix` in Nixpkgs."
+
+    else if pkg.meta.broken
+    then null
+
+    else makePkgConfigTest moduleName pkg;
+
+  makePkgConfigTest = moduleName: pkg: runCommand "check-pkg-config-${moduleName}" {
+    nativeBuildInputs = [ pkg-config ];
+    buildInputs = [ pkg ];
+    inherit moduleName;
+    meta = {
+      description = "Test whether ${pkg.name} exposes pkg-config module ${moduleName}";
+    }
+    # Make sure licensing info etc is preserved, as this is a concern for e.g. cache.nixos.org,
+    # as hydra can't check this meta info in dependencies.
+    # The test itself is just Nixpkgs, with MIT license.
+    // builtins.intersectAttrs
```

Comment:
I'd just update the `meta` of the source package instead which seems more comprehensible – hydra will behave the same for both (i.e. `pkg.meta // { description = … ; }`). The new `meta` set is run through `mkDerivation` anyways.

Reply:
It isn't equivalent to the original package though, so I'd rather be conservative in terms of what we inherit. For example, the description would be wrong. For instance, your suggestion would copy the `longDescription`, which is wrong.